docs/defaults-detailed.rst

```diff
@@ -9,169 +11,640 @@ them.
+pages to make comparsion between :file:`/etc/network/interfaces` configuration
+and ``debops.ifupdown`` configuration easier. Examples are verbose to reflect
+the examples from the wiki page, but some of the parameters can be omitted to
+let the role autogenerate them.
+
+Keep in mind that the ``auto`` parameter, included in the examples for
+completeness, usually should be avoided in the newer OS releases (Jessie+,
+Trusty+) on ``systemd``-based hosts. This is done so that the additional
+processes related to a given network interfaces are put in their own
+``ifup@.service`` cgroup instead of being grouped together under
+``networking.service`` cgroup.
+
+Use DHCP and SLAAC to `automatically configure the network interface <https://wiki.debian.org/NetworkConfiguration#Using_DHCP_to_automatically_configure_the_interface>`_:
+
+.. code-block:: yaml
+
+   ifupdown__interfaces:
+     'eth0':
+       auto: True
+       allow: 'hotplug'
+       inet: 'dhcp'
+       inet6: 'auto'
+
+`Configure the network interface manually <https://wiki.debian.org/NetworkConfiguration#Configuring_the_interface_manually>`_
+using static IPv4 and IPv6 configuration:
+
+.. code-block:: yaml
+
+   ifupdown__interfaces:
+     'static-eth0':
+       iface: 'eth0'
+       auto: True
+       inet: 'static'
+       inet6: 'static'
+       addresses: [ '192.0.2.7/24', '2001:db8::c0ca:1eaf/64' ]
+       gateway4: '192.0.2.254'
+       gateway6: '2001:db8::1ead:ed:beef'
+
+Configure an interface `without an IP address <https://wiki.debian.org/NetworkConfiguration#Bringing_up_an_interface_without_an_IP_address>`_:
+
+.. code-block:: yaml
+
+   ifupdown__interfaces:
+
+     'eth0':
+       inet: 'manual'
+       options: |
+         pre-up ifconfig $IFACE up
+         post-down ifconfig $IFACE down
+
+    'eth0.99':
+      inet: 'manual'
+      options: |
+        post-up ifconfig $IFACE up
+        pre-down ifconfig $IFACE down
+
+Configure `DNS nameservers and search domains <https://wiki.debian.org/NetworkConfiguration#The_resolvconf_program>`_
+with an autogenerated default interface:
+
+.. code-block:: yaml
 
-``port``
-  Name of other network interface which this interface might depend on. Usage
-  depends on the interface type.
+   ifupdown__interfaces:
+     'external':
+       iface: '{{ ifupdown__external_interface }}'
+       inet: 'dhcp'
+       dns_nameservers: [ '12.34.56.78', '12.34.56.79' ]
+       dns_search: 'example.com'
 
-  In ``bridge`` interfaces, you can specify the name or list of interfaces to add
-  to the bridge.
+Configure `static bridge <https://wiki.debian.org/NetworkConfiguration#Bridging>`_
+between two Ethernet interfaces:
 
-  In ``vlan`` interfaces, you can define the name of the interface used as
-  ``vlan_raw_device``.
+.. code-block:: yaml
+
+   ifupdown__interfaces:
+
+     'eth0':
+       inet: 'manual'
+       inet6: False
+
+     'eth1':
+       inet: 'manual'
+       inet6: False
```

Comment:
If `inet` is not defined, role will add it with the value that depends on the interface type (Ethernet has `dhcp`, bridge and bonding has `manual`).

If `inet6` is not defined, Ethernet interfaces will get `inet6: 'auto'` by default to get stateless configuration.